fabtests/functional/rdm_stress.c

```diff
@@ -593,58 +592,280 @@ static int run_child(void)
+			(*idx)++; /* advance to value token */
+			cur_token = &token[*idx];
+			if (!get_uint64_val(json_payload, cur_token, &ctrl->offset))
+				goto err_out;
+		} else {
+			goto err_out;
+		}
+	}
+
+	/* op is rquired for rpc_ctrl to be valid */
+	if (ctrl->op == 0xffffffff)
+		goto err_out;
+
+	return true;
+
+err_out:
+	printf("Invalid JSON entry: %.*s\n",
+		token[obj_idx].end - token[obj_idx].start,
+		&json_payload[token[obj_idx].start]);
+	init_rpc_ctrl(ctrl); /* in case this is the last token */
+
+	return false;
+}
+
+/* read and parse control file */
+static int init_ctrls(const char *ctrlfile)
+{
+	FILE *ctrl_f;
+	struct stat statbuf;
+	char *json_payload;	/* control file loaded in string */
+	int num_tokens;		/* # of JSON tokens in the control file */
+	int num_obj;		/* # of JSON objects = possible rpc_ctrl entries */
+	jsmn_parser parser;
+	jsmntok_t *tokens;
+	int start, i;
+	int ret = 0;
+
+	ctrl_f = fopen(ctrlfile, "r");
+	if (!ctrl_f)
+		return -errno;
+
+	if (stat(ctrlfile, &statbuf))
+		return -errno;
+
+	json_payload = malloc(statbuf.st_size + 1);
+	if (!json_payload) {
+		ret = -FI_ENOMEM;
+		goto no_mem_out;
+	}
+
+	if (fread(json_payload, statbuf.st_size, 1, ctrl_f) != 1) {
+		ret = -FI_EINVAL;
+		goto read_err_out;
+	}
+	json_payload[statbuf.st_size] = 0;
+
+	/* get # of tokens, allcoate memory and parse JSON */
+	jsmn_init(&parser);
+	num_tokens = jsmn_parse(&parser, json_payload, statbuf.st_size, NULL, 0);
+	if (num_tokens < 0) {
+		ret = -FI_EINVAL;
+		goto read_err_out;
+	}
+
+	tokens = malloc(sizeof(jsmntok_t) * num_tokens);
+	if (!tokens) {
+		ret = -FI_ENOMEM;
+		goto read_err_out;
+	}
+
+	jsmn_init(&parser);
+	if (jsmn_parse(&parser, json_payload, statbuf.st_size, tokens, num_tokens) != num_tokens) {
+		ret = -FI_EINVAL;
+		goto parse_err_out;
+	}
+
+	/* find the first JSON array bypassing comments at the beginning of control file */
+	for (start = 0; start < num_tokens && tokens[start].type != JSMN_ARRAY; start++)
+		;
+	if (start == num_tokens) {
+		ret = -FI_EINVAL;
+		goto parse_err_out;
+	}
+
+	/* count # of JSMN_OBJECT which is # of potential rpc_ctrl entries */
+	for (i = start, num_obj = 0; i < num_tokens; i++)
+		if  (tokens[i].type == JSMN_OBJECT)
+			num_obj++;
+
+	if (num_obj <= 0) {
+		ret = -FI_EINVAL;
+		goto parse_err_out;
+	}
+
+	ctrls = malloc(sizeof(struct rpc_ctrl) * num_obj);
+	if (!ctrls) {
+		ret = -FI_ENOMEM;
+		goto parse_err_out;
+	}
+
+	/* extract rpc_ctrl structs from tokens */
+	for (ctrl_cnt = 0; start < num_tokens; start++) {
+		if (tokens[start].type != JSMN_OBJECT)
+			continue;
+
+		if (add_ctrl(json_payload, num_tokens, tokens, &ctrls[ctrl_cnt], &start))
+			ctrl_cnt++;
+	}
+
+	free(tokens);
+	free(json_payload);
+	fclose(ctrl_f);
+
+	if (ctrl_cnt <= 0) {
+		free(ctrls);
+		ctrls = NULL;
+		return -FI_EINVAL;
+	}
+
 	return 0;
+
+parse_err_out:
+	free(tokens);
+
+read_err_out:
+	free(json_payload);
+
+no_mem_out:
+	fclose(ctrl_f);
+
+	return ret;
```

Comment:
nit: remove extra blank lines in the error path

Reply:
I can remove the lines between labels.  In general I'd like to call out returns, but i can follow the coding style in the file.